baseapp/queryrouter.go

```diff
@@ -15,7 +15,7 @@ var _ sdk.QueryRouter = NewQueryRouter()
 // NewQueryRouter returns a reference to a new queryRouter.
 //
 // TODO: Either make the function private or make return type (queryRouter) public.
-func NewQueryRouter() *queryRouter { // nolint: golint
+func NewQueryRouter() *queryRouter { //nolint:golint
```

Comment:
```suggestion
func NewQueryRouter() *queryRouter {
```

Reply:
That would cause golint to output:
```
exported func NewQueryRouter returns unexported type *baseapp.queryRouter, which can be annoying to use
```
And it would be correct.
There is a readme above for that i think

Reply:
can you make `queryRouter` public (i.e rename to `QueryRouter`) and remove the `TODO:` ? It's only used for a private field from `Baseapp`:

`queryRouter:    NewQueryRouter(),`

Reply:
Done in: b1c2350

Reply:
Wait. That would make more sense if it become private, not public. Since it is used only in private field and is exposed via interface (`sdk.Router`)?